app/src/main/java/com/example/plantsapp/presentation/ui/MainActivity.kt

```diff
@@ -48,18 +50,36 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
+    override fun onDestroy() {
+        super.onDestroy()
+
+        supportFragmentManager.unregisterFragmentLifecycleCallbacks(
+            bottomBarVisibilityCallback
+        )
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                supportFragmentManager.popBackStack()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
     private fun openFragment(fragment: Fragment) {
         supportFragmentManager.popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE)
         supportFragmentManager.commit {
             replace(R.id.fragment_container, fragment)
         }
     }
 
-    override fun onDestroy() {
-        super.onDestroy()
-
-        supportFragmentManager.unregisterFragmentLifecycleCallbacks(
-            bottomBarVisibilityCallback
-        )
+    private fun setUpHomeButtonOnAppBar() {
+        supportFragmentManager.addOnBackStackChangedListener {
+            supportActionBar?.setDisplayHomeAsUpEnabled(
+                supportFragmentManager.backStackEntryCount > 0
```

Comment:
why not just true? (maybe there are some things that I did-not noticed, so because of that im asking)

Reply:
> why not just true? (maybe there are some things that I did-not noticed, so because of that im asking)

so we won't have this back button on Plants Screen, Tasks Screen and Profile Screen, where it is not needed
and that was in requirements
![Screen Shot 2021-12-23 at 2 40 05 PM](https://user-images.githubusercontent.com/43578259/147235406-8476ce0a-a5dc-4367-90ab-e7f61e212418.png)

Reply:
maybe just add 
val shouldShowHomeButton = supportFragmentManager.backStackEntryCount > 0
and then 
supportActionBar?.setDisplayHomeAsUpEnabled(shouldShowButton)